name: Tests
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        python: ["3.7", "3.8", "3.9", "3.10", "3.11"]
        plone: ["6.0-latest", "5.2-latest"]
        exclude:
          - plone: "5.2-latest"
            python: "3.9"
          - plone: "5.2-latest"
            python: "3.10"
          - plone: "5.2-latest"
            python: "3.11"
          - plone: "6.0-latest"
            python: "3.7"

    steps:
      # git checkout
      - uses: actions/checkout@v2

      - name: Setup Plone ${{ matrix.plone }} with Python ${{ matrix.python }}
        id: setup
        uses: plone/setup-plone@v2.0.0
        with:
          python-version: ${{ matrix.python }}
          plone-version: ${{ matrix.plone }}

      - name: Install package
        # FIXE: plone.app.transmogrifier 3.0.0 has an error in Plone 6. See:
        # https://github.com/collective/plone.app.transmogrifier/pull/32
        # This bug has been fixed. So, as soon as a new version is released,
        # we must remove this installation through github.
        run: |
          pip install git+https://github.com/collective/plone.app.transmogrifier.git
          pip install -e ".[test]"

      # test
      - name: test
        run: |
          zope-testrunner --auto-color --auto-progress --test-path src
